ci: Build documentation in pipeline

NicolasT opened the pull request:
Making sure the documentation can be built at any point in time is useful. Let's enforce this in CI.

The infrastructure was borrowed from the MetalK8s project (ref d9c18547f3e2240bada58db5126ecce8d0d58fa5).

Also, enable the `sphinxcontrib-spelling` extension.

See: https://github.com/scality/metalk8s

NicolasT committed:
docs: Enable `sphinxcontrib-spelling`

NicolasT committed:
ci: Build documentation in pipeline

Making sure the documentation can be built at any point in time is
useful. Let's enforce this in CI.

The infrastructure was borrowed from the MetalK8s project (ref
d9c18547f3e2240bada58db5126ecce8d0d58fa5).

See: https://github.com/scality/metalk8s

bert-e commented:
# Hello NicolasT,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/docs-build` has diverged from
`development/1.0` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/1.0` into `improvement/docs-build`
* Rebase `improvement/docs-build` onto `origin/development/1.0`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

NicolasT committed:
Merge branch 'development/1.0' into improvement/docs-build

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @NicolasT

* @giacomoguiulfo

* @vrancurel

* @ssalaues

NicolasT commented:
Hmh, odd CI failures... Maybe because the version of Eve running on this repository is too old. Any insights, @tcarmet ?

ssalaues commented on eve/main.yml:
```diff
@@ -104,11 +104,31 @@ stages:
+      path: eve/workers/doc-builder.yaml
+      images:
+        doc-builder:
+          context: '.'
```
For some reason it looks like Eve is having issues with reading the `context`

tcarmet replied:
yeah my bad, there's an old version of eve on zenko. I'm updating it!

tcarmet replied:
update done! Working as expected now :)

NicolasT replied:
Thanks @tcarmet!

NicolasT committed:
Merge branch 'development/1.0' into improvement/docs-build

NicolasT committed:
docs: Make build succeed

These changes are not necessarily *desired* changes. However, we need
some baseline non-broken build...

See: #340
See: https://github.com/scality/Zenko/pull/340

NicolasT committed:
docs: Exclude GKE deployment doc

This makes LaTeX rendering fail somehow.

NicolasT committed:
docs: Ensure PDF can be built

Special handling of SVG images is required.

NicolasT commented:
I made some changes to the actual docs to make the tree actually build.

These may not be the changes @wabernatScality actually wants, so may certainly need to be fixed up later.

However! Despite this, I'd argue getting this integrated and ensuring any later edits actually build is more important than keeping non-building docs around.

smaffulli commented:
Absolutely, we have to have this. I just pulled the latest development/1.0 branch and docs don't build cleanly.

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`



The following branches will **NOT** be impacted:

* `development/0.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of Release Engineering will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of Release Engineering now.

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`



The following branches have **NOT** changed:

* `development/0.9`



Please check the status of the associated issue None.

Goodbye NicolasT.
